com.playeveryware.eos/Documentation~/unit-testing.md

```diff
@@ -0,0 +1,148 @@
+# Unit Testing Guide for EOS Plugin for Unity
+
+## Introduction to Unit Testing
+
+Unit testing is a software testing technique in which individual components of the code are tested to ensure they work as expected. Each unit test should focus on a single function, class, or method and provide immediate feedback on whether that piece of code is functioning correctly. The goal is to catch and address bugs early in development, ensuring a stable and maintainable codebase.
+
+### Benefits of Unit Testing
+- **Improved code quality:** Unit tests ensure that code functions as expected before it is merged into the main codebase.
+- **Early bug detection:** Tests provide an opportunity to catch and fix issues during development.
+- **Facilitates refactoring:** With a solid unit test suite, developers can confidently refactor code knowing tests will highlight regressions.
+- **Code coverage:** Tests help measure how much of the codebase is covered, ensuring critical functionality is always tested.
+
+## Testing Frameworks in Use
+
+Historically, this project has used both **NUnit** and the **Unity Test Framework** for unit testing. However, moving forward, we are transitioning to using **NUnit** exclusively. NUnit offers more flexibility and integration options for broader test case management and is better suited for the long-term needs of this project.
+
+While there are still some existing tests using the Unity Test Framework, all new tests should be written using **NUnit**. This will ensure consistency across the test suite and streamline the development process.
+
+### Why NUnit?
+
+- **Flexibility:** NUnit supports a wide range of assertions, test case parameters, and test fixtures.
+- **CI/CD Compatibility:** NUnit can be easily integrated into Continuous Integration and Deployment pipelines (planned for future).
+- **Broad usage:** NUnit is a widely-adopted framework, making it familiar to many developers and well-supported across different IDEs.
+
+## Scope of Unit Testing
+
+Unit tests should be written for **scripts used in the sample scenes** and for the **core plugin code**. The following guidelines outline the scope:
+
+### What Should Be Tested:
+- Core plugin functionality (e.g., helper classes, business logic, utilities, etc.).
+- Scripts attached to game objects or components in the sample scenes (but not the scenes themselves).
+
+### What Should Not Be Tested:
+- The **EOS SDK** itself. The SDK’s behavior is outside the scope of this project’s unit tests.
```

Comment:
We do have some tests for Interfaces. Is that considered something we don't want to be testing?

Reply:
We should change this to indicate that unit testing of the EOS SDK C# wrapper is not a current _priority_. It's not that it should _not_ be tested, just that it is not the current priority. Once the language has been updated to reflect that, I will mark this conversation as resolved.